File: Daily Test/Python/Matrix - Detect Loop 14.2.2026.py
r, c = map(int, input().split())
matrix = [list(map(int, input().split())) for _ in range(r)]
i, j = map(int, input().split())
i -= 1
j -= 1

visited = set()

while True:
    if i < 0 or i >= r or j < 0 or j >= c:
        print("NO")
        break

    if (i, j) in visited:
        print("YES")
        break

    visited.add((i, j))

    val = matrix[i][j]

    if val >= 0:
        if val % 2 == 0:
            i -= 1
        else:
            j += 1
    else:
        if (-val) % 2 == 0:
            i += 1
        else:
            j -= 1

File: Daily Test/C++/Matrix - Detect Loop 14.2.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int r, c;
    cin >> r >> c;

    vector<vector<int>> matrix(r, vector<int>(c));
    for(int i = 0; i < r; i++)
        for(int j = 0; j < c; j++)
            cin >> matrix[i][j];

    int i, j;
    cin >> i >> j;
    i--; 
    j--;

    set<pair<int,int>> visited;

    while(true) {
        if(i < 0 || i >= r || j < 0 || j >= c) {
            cout << "NO";
            break;
        }

        if(visited.count({i,j})) {
            cout << "YES";
            break;
        }

        visited.insert({i,j});
        int val = matrix[i][j];

        if(val >= 0) {
            if(val % 2 == 0) i--;
            else j++;
        } else {
            if((-val) % 2 == 0) i++;
            else j--;
        }
    }
}